docs/recipes/recipes.md

```diff
@@ -477,6 +477,99 @@ int main(int argc, char **argv)
 ```
 </details>
 
+## Custom widgets
+
+### Custom Button
+
+```slint
+Button := Rectangle {
+    property text <=> txt.text;
+    callback clicked <=> touch.clicked;
+    border-radius: height / 2;
+    border-width: 1px;
+    border-color: background.darker(25%);
+    background: touch.pressed ? #6b8282 : touch.has-hover ? #6c616c :  #456;
+    height: txt.preferred-height * 1.33;
+    min-width: txt.preferred-width + 20px;
+    txt := Text {
+        x: (parent.width - width)/2 + (touch.pressed ? 2px : 0);
+        y: (parent.height - height)/2 + (touch.pressed ? 1px : 0);
+        color: touch.pressed ? #fff : #eee;
+    }
+    touch := TouchArea { }
+}
+
+Demo := Window {
+    VerticalLayout {
+        alignment: start;
+        Button { text: "Button"; }
+    }
+}
+```
+
+### CheckBox
+
+```
+export CheckBox := Rectangle {
```

Comment:
Since the visual representation of this widget is closer to a physical switch instead of a box with a check mark, I suggest to call this a "ToggleSwitch", "Toggle" or a "Switch". My vote is on the first one :)